Panic when parsing separate ClusterConfig in same file

Docs specify providing ClusterConfig separately in the same file:
https://github.com/k0sproject/k0sctl/tree/main#speck0sconfig-mapping-optional-default-auto-generated
But doing this causes panic due to Config not being initialized. 

```
k0sctl apply - <<EOF                       
apiVersion: k0sctl.k0sproject.io/v1beta1
kind: Cluster
spec:
  hosts:
    - role: single
      ssh:
        address: 10.0.0.1
---
apiVersion: k0s.k0sproject.io/v1beta1
kind: ClusterConfig
metadata:
  name: my-k0s-cluster
spec:
  api:
    externalAddress: 10.0.0.2
EOF
```

Causes at commit bedaa91

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/k0sproject/dig.Mapping.Merge(0x0, 0x140003c5050, {0x0, 0x0, 0x0})
        github.com/k0sproject/dig@v0.4.0/mapping.go:203 +0x3d0
github.com/k0sproject/k0sctl/cmd.readConfig(0x14000390dc0?)
        github.com/k0sproject/k0sctl/cmd/flags.go:285 +0x5d0
github.com/k0sproject/k0sctl/cmd.initManager(0x14000390700)
        github.com/k0sproject/k0sctl/cmd/flags.go:313 +0x20
github.com/k0sproject/k0sctl/cmd.init.actions.func15(0x14000390700)
        github.com/k0sproject/k0sctl/cmd/flags.go:151 +0x5c
github.com/urfave/cli/v2.(*Command).Run(0x105a91140, 0x14000390700, {0x1400009f4e0, 0x1, 0x1})
        github.com/urfave/cli/v2@v2.27.7/command.go:216 +0x388
github.com/urfave/cli/v2.(*Command).Run(0x14000163a20, 0x14000390540, {0x140000a2080, 0x2, 0x2})
        github.com/urfave/cli/v2@v2.27.7/command.go:269 +0x774
github.com/urfave/cli/v2.(*App).RunContext(0x1400026aa00, {0x1053a8bc0, 0x105ae3aa0}, {0x140000a2080, 0x2, 0x2})
        github.com/urfave/cli/v2@v2.27.7/app.go:333 +0x4c8
github.com/urfave/cli/v2.(*App).Run(0x14000017f38?, {0x140000a2080?, 0x0?, 0x0?})
        github.com/urfave/cli/v2@v2.27.7/app.go:307 +0x3c
main.main()
        ./main.go:17 +0x35c
```

--- a/cmd/flags.go
+++ b/cmd/flags.go
@@ -275,6 +275,9 @@ func readConfig(ctx *cli.Context) (*v1beta1.Cluster, error) {
 		return nil, fmt.Errorf("failed to unmarshal cluster config: %w", err)
 	}
 	if k0sConfigs, err := mr.GetResources("k0s.k0sproject.io/v1beta1", "ClusterConfig"); err == nil && len(k0sConfigs) > 0 {
+		if cfg.Spec.K0s.Config == nil {
+			cfg.Spec.K0s.Config = make(dig.Mapping)
+		}
 		for _, k0sConfig := range k0sConfigs {
 			k0s := make(dig.Mapping)
 			log.Debugf("unmarshalling %d bytes of config from %v", len(k0sConfig.Raw), k0sConfig.Filename())

